algebra/src/msm/variable_base.rs

```diff
@@ -5,6 +5,16 @@ use num_traits::One;
 pub struct VariableBaseMSM;
 
 impl VariableBaseMSM {
+    /// Multi-scalar multiplication (MSM), based on the bucket method of
+    /// [Yao's 1976 paper](https://www.ii.uni.wroc.pl/~aje/WordEq2015/papers/addition_chains_Yao.pdf).
+    /// This implementation uses affine arithmetics and batch inversion. 
+    /// Given base points `G[i]` together with scalars `c[i]`, `i=1,..n`, we write 
+    ///    c[i] = sum  c[i,j] * D^j,
+    ///    s[j] = Sum_i c[i,j] * G[i],
+    /// for each order `j`. 
+    /// The final result of the multi-scalar multiplication is then combined by 
+    ///    s = Sum_j  A^j * s[j], 
+    /// using double-and-add.
```

Comment:
To me, this shortened explanation hides that the bucket method is applied to the digits from `D`. 
How about 
``` 
/// Given base points `G[i]` together with scalars `c[i]`, `i=1,..n`, we write 
///    c[i] = sum  c[i,j] * D^j,
/// with digits `c[i,j]` from the interval `{0,...,D-1}`, and compute 
///    s[j] = Sum_i c[i,j] * G[i],
/// for each order `j` using the Pippenger-Yao bucket method.
```

Reply:
Fine thanks !